bsip-0069.md

```diff
@@ -0,0 +1,165 @@
+
+## Proposed new predicates:
+
+We propose the addition of the following new predicates, organized by purpose below:
+
+### Extending account and asset creation safety:
+
+To further support account and asset creation safety, we propose to add the following predicates:
+
+| Predicate | Description |
+|:----------|:------------|
+| `account_authorities_match_predicate` | Tests that account authorities (owner, active, special, etc.) are as supplied in predicate|
+| `asset_issuer_is_account_predicate` | Tests asset issuer |
+
+Although the existing predicates already allow to check for the successful creation of a new account or asset by testing the correlation of an object ID and a name string, this still leaves open a vulnerability from an attacking party trying to register the same account name or asset symbol.  This would be a targetted attack, in which an attacker would have to observe an attempt to register a particular name and then front-run a competing registration operation that “steals” the name.  Although tricky to implement, such an attack is certainly possible.  For a registration script wishing to quickly fund a newly-created account, a better way to test successful creation would be to test that the account authorities are as expected, rather than just the correlation of account name to id. Similarly for asset creation, a test of asset issuer offers assurance that the asset created is in fact in control of the expected issuer.
+
+### Head block time predicate:
+
+We propose to add the following predicate, which supports applications such as payment channels:
+
+| Predicate | Description |
+|:----------|:------------|
+| `head_block_time_ge_time_predicate` | Head block time meets or exceeds `time`. Ensures a transaction is only valid at or after a specified time. |
+
+This predicate allows for the signing of future-dated transactions (up to the `maximum_time_until_expiration` limit set by committee).  Payment channel constructs, such as proposed in [BSIP-0063](bsip-0063.md), can use this to sign refund transctions to be used if a counterparty fails to meet an obligation to close out a channel within an agreed upon time window.
```

Comment:
The first sentence is wrong. Perhaps instead
> This predicate allows for the pre-signing of future-dated transactions (within the limits of the TAPOS range, i. e. 65k blocks).

Reply:
Interesting; I had assumed `maximum_time_until_expiration` would be w.r.t. the referenced TaPoS block.  But now finding the code [here](https://github.com/bitshares/bitshares-core/blob/e8567d0425ec27d29036a9a9178df5afd8ca6f25/libraries/chain/db_block.cpp#L668:L672) it does seem it's w.r.t. the head_block_time at the time the transaction is seen by the network and applied to a block.  This actually extends the usable life of a payment channel.  Great!

Thanks @pmconrad.  Will revise.

As an interesting aside:  This actually provides a way to have an implicit head_block_time assertion without actually having the head_block_time assert predicate:

One could simply create a transaction with an expiration time of `maximum_time_until_expiration` + `intended_validity_time`, and then that transaction could NOT be included in a block until `intended_validity_time` arrives.  (Still subject to the TaPoS constraint, of course.)

**Q: Does this negate the value of having the predicate explicitly available?**   I think it's still valuable to have, for the clarity it provides.  (Should make it easier for a wallet managing a payment channel to validate the construction of a refund transaction.)  Also it provides protection against the chance of the `maximum_time_until_expiration` parameter changing within the lifetime of the channel and therefore messing with the trigger times, which I think is important.  But aside from that risk... it does now seem that the predicate duplicates an existing (implicit) capability.  Thoughts?

Reply:
`maximum_time_until_expiration` is configurable by the committee. The implicit assertion you describe is therefore not reliable, so the proposed new assertion is an actual improvement.